ddl/partition.go

```diff
@@ -1184,14 +1185,30 @@ func buildRangePartitionDefinitions(ctx sessionctx.Context, defs []*ast.Partitio
 
 		buf := new(bytes.Buffer)
 		// Range columns partitions support multi-column partitions.
-		for _, expr := range clause.Exprs {
+		for i, expr := range clause.Exprs {
 			expr.Accept(exprChecker)
 			if exprChecker.err != nil {
 				return nil, exprChecker.err
 			}
-			expr.Format(buf)
-			piDef.LessThan = append(piDef.LessThan, buf.String())
-			buf.Reset()
+			if len(partValStrings) > i {
+				partVal := partValStrings[i]
+				switch colTypes[i].EvalType() {
+				case types.ETInt:
+					// no wrapping
+				case types.ETDatetime, types.ETString:
+					if _, ok := clause.Exprs[i].(*ast.MaxValueExpr); !ok {
+						// Don't wrap MAXVALUE
+						partVal = driver.WrapInSingleQuotes(partVal)
```

Comment:
Why do this?

Reply:
Since the other implementations are using the `expr.Format()` (line 1208) to format a string, and that will wrap in in single quotes ([see here](https://github.com/pingcap/tidb/blob/f89609d018c5ec2a884212b6bbc653f78ec98c3a/types/parser_driver/value_expr.go#L170)).